packages/connections/src/XHRConnection.js

```diff
@@ -0,0 +1,170 @@
+import AbstractConnection from "./AbstractConnection";
+import ConnectionEvent from "./ConnectionEvent";
+
+export const ALLOWED_METHODS = ["GET", "POST", "PUT", "DELETE"];
+
+/**
+ * Basic XHR Connection
+ * @todo add/remove headers
+ */
+export default class XHRConnection extends AbstractConnection {
+  /**
+   * Initializes an Instance of XHRConnection
+   * @constructor
+   * @param {string} url – URL to be fetched
+   * @param {object} [options]
+   * @param {string} [options.method=GET] – used method
+   * @param {*} [options.body] – payload for request
+   * @param {object} [options.headers] – additional headers (like content-type)
+   * @param {string} [options.responseType]
+   */
+  constructor(url, options = {}) {
+    super(url);
+
+    const {
+      method = "GET",
+      body = null,
+      headers = {},
+      responseType = "json"
+    } = options;
+
+    if (ALLOWED_METHODS.includes(method) === -1) {
+      throw new Error("Invalid method for XHR Request.");
+    }
+
+    /**
+     * @property {string}
+     * @protected
+     * @name XHRConnection#method
+     */
+    Object.defineProperty(this, "method", { value: method });
+
+    /**
+     * @property {*}
+     * @protected
+     * @name XHRConnection#body
+     */
+    Object.defineProperty(this, "body", { value: body });
+
+    /**
+     * @property {object}
+     * @protected
+     * @name XHRConnection#headers
+     */
+    Object.defineProperty(this, "headers", { value: headers });
+
+    /**
+     * @property {string}
+     * @protected
+     * @name XHRConnection#responseType
+     */
+    Object.defineProperty(this, "responseType", { value: responseType });
+
+    /**
+     * @property {XMLHttpRequest}                 ̵
+     * @protected
+     * @name XHRConnection#xhr
+     */
+    Object.defineProperty(this, "xhr", { value: new XMLHttpRequest() });
+  }
+
+  get response() {
+    return this.xhr.response;
+  }
+
+  get status() {
+    return this.xhr.status;
+  }
+
+  /**
+   * create, prepare, open and send the xhr request
+   * @param {string} [token] – authentication token
+   */
+  open(token) {
+    super.open(token);
```

Comment:
We don't need to pass token to super. Please see other related comment on `AbstractConnection`.

Reply:
We should always pass all defined arguments to the super class, even if they'er not used in the super class to allow for future improvements and advancements.

Reply:
minor #nit but passing `...arguments` will be more future proof and require less maintenance.

Reply:
After making this package standalone we should consider adding a simple hooks system again. We discarded it before because we were only thinking about the CM, but when these connections could potentionally used by anything (even without the CM), we should make the header injecion "more open".

As a first step I'd just add `addHeader` & `removeHeader` methods, everything else should be held back for a future "v2".